garak/probes/base.py

```diff
@@ -136,6 +135,8 @@ def _mint_attempt(self, prompt, seq=None) -> garak.attempt.Attempt:
         new_attempt.status = garak.attempt.ATTEMPT_STARTED
         new_attempt.goal = self.goal
         new_attempt.seq = seq
+        if prompt is not None:
+            new_attempt.prompt = prompt
```

Comment:
What will the value of `new_attempt.prompt` be if prompt is None? Won't it also be None? I'm not sure this change is doing anything. Should we raise something here instead?

Reply:
`None` is the default value for `Attempt()`, would it be clearer to refactor this whole method to something like:
``` Python
    def _mint_attempt(self, prompt=None, seq=None) -> garak.attempt.Attempt:
        new_attempt = garak.attempt.Attempt(
            "prompt" = prompt,
            "probe_classname" = (
                str(self.__class__.__module__).replace("garak.probes.", "")
                + "."
                + self.__class__.__name__
            ),
            "status" = garak.attempt.ATTEMPT_STARTED,
            "goal" = self.goal
            "seq" = seq
        )
        new_attempt = self._attempt_prestore_hook(new_attempt, seq)
        return new_attempt
```

Reply:
`None` prompts are now not valid. Setting `prompt` triggers building a conversation data structure behind the scenes. Doing `attempt.prompt = None` translates to building a single-turn invalid dialogue `[{"role":"user", "content":None}]`, which I don't think we want to allow; blank prompts are represented with empty string `""`. I've made this constraint clearer in `garak.attempt.Attempt` which now raises `TypeError` if one tries to set `attempt.prompt` to `None`.

It should however be possible to construct an `Attempt` without specifying a value for `prompt`, which is the represented in the change in order of operations & the `is None` check seen here. In that case, one can defer setting `attempt.prompt` and just instantiate an `Attempt` without a prompt. This is used where we want an `attempt` object, e.g. for its `uuid`, _before_ the initial prompt text is known.

The requirement to be able to mint an attempt without knowing the prompt led to the logic here: `_mint_attempt` should only set a prompt, and thus invoke the behind-the-scenes orchestration, if a prompt is actually given.

We can refactor `_mint_attempt()` to use the constructor, yeah, will do.

Reply:
Oh right, that's now possible. OK!